src/interactive/Dropdown.jsx

```diff
@@ -71,27 +113,38 @@ class Dropdown extends React.PureComponent {
 			ref => !ref.contains(e.target)
 		);
 
-		if (isNotDropdownClick) {
+		if (isNotDropdownClick || (!isNotDropdownClick && this.props.closeOnBodyClick)) {
 			this.closeContent(e);
 		}
-	}
+	};
 
-	onBodyKeyDown(e) {
+	onBodyKeyDown = (e: SyntheticKeyboardEvent<*>) => {
 		if (e.key === 'Escape') {
-			this.closeContent();
+			this.closeContent(e);
 		}
-	}
+	};
 
 	componentDidMount() {
-		document.body.addEventListener('click', this.onBodyClick);
-		document.body.addEventListener('keydown', this.onBodyKeyDown);
+		if (document.body) {
```

Comment:
curious about this guard - I heard mention there was an error related to this, but not sure how `document.body` would not exist in `componentDidMount` - any info?